File: DSA/saumya/Day14_Q1.py
def min_moves(height):
    moves = 0
    while True:
        removed = False
        new_height = [height[0]]
        for i in range(1, len(height)):
            if height[i] >= new_height[-1]:
                new_height.append(height[i])
            else:
                removed = True
        if not removed:
            break
        height = new_height
        moves += 1
    return moves

n = 3
height = [1, 2, 1]
print(min_moves(height))  # Output: 1

File: DSA/Ishita-Nigam/Day14/Day14_Q1.java

import java.util.Scanner;

public class Day14_Q1 {

    public static int minMoves(int[] height) {
        int n = height.length;
        int moves = 0;
        
        while (true) {
            boolean removed = false;
            for (int i = 1; i < n; i++) {
                if (height[i - 1] > height[i]) {
                    for (int j = i; j < n; j++) {
                        height[j - 1] = height[j];
                    }
                    n--;
                    removed = true;
                    break;
                }
            }
            if (!removed) {
                break;
            }
            moves++;
        }
        
        return moves;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter the number of students: ");
        int n = scanner.nextInt();
        
        int[] height = new int[n];
        System.out.println("Enter the heights of the students: ");
        for (int i = 0; i < n; i++) {
            height[i] = scanner.nextInt();
        }
        
        int result = minMoves(height);
        System.out.println("Minimum number of moves: " + result);
        
        scanner.close();
    }
}
